app/exporters/hathi/content_package.rb

```diff
@@ -139,8 +132,20 @@ def image_file
     end
 
     class DerivativePage < Page
-      def image_file
-        @fileset.jp2_derivative
+      def image_filename
+        "#{@basename}.jp2"
+      end
+
+      def path_to_file
+        @path_to_file ||= Pathname.new(jp2_derivative.path)
+      end
+
+      def jp2_derivative
+        @jp2_derivative ||= JP2Creator.new(filename: original_file_path.to_s).generate
```

Comment:
What will be the impact on the creation of these packages now? I assume they will take much longer to create? I.e. should we ask @cwulfman to test it with this branch?

Reply:
Yes, they'll definitely take longer to create, but I also don't think we're actively generating these right now. I'd be disinclined to hold this up for that, but @cwulfman should definitely try stuff out again if he's still working on this.

Reply:
Yes, they'll definitely take longer to create, but I also don't think we're actively generating these right now. I'd be disinclined to hold this up for that, but @cwulfman should definitely try stuff out again if he's still working on this.

Reply:
👍